### PR TITLE
Feature/week9 homework

### PR DESCRIPTION
**Pull request by d-ben7**
Hi Fuad.

I'm aware the code is quite messy in some areas. I will clean it up soon.

### CODE REVIEW

File: Homework/Week9/app/src/test/java/com/android/pokedex/viewmodel/DetailViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        detailViewModel = DetailViewModel(repository, pokeId)
+    }
+
+    @Test
+    fun getSelectedPokemon() {
+        val data = MutableLiveData(bulbasaur)
+        whenever(detailViewModel.getSelectedPokemon()).thenReturn(data)
+        detailViewModel.getSelectedPokemon().observeForever(observer)
+
+        verify(observer).onChanged(bulbasaur)
```
Comment (filbabic) [inline Homework/Week9/app/src/test/java/com/android/pokedex/viewmodel/DetailViewModelTest.kt:51]: These are really nice! Good job on verifying the behavior & asserting the values.

I'd maybe go one extra mile where you'd (as in the presented solution), verify interactions between layers and that there are _no more interactions_ afterwards!

File: Homework/Week9/app/src/test/java/com/android/pokedex/viewmodel/FavoritesViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    @Test
+    fun getFavorites() {
+        val pikachu = Pokemon(25, "Pikachu", "Eletric", 1.1, 1.1, true)
+        val bulbasaur = Pokemon(1, "Bulbasaur", "Grass", 1.1, 1.1, true)
+        val charmander = Pokemon(5, "Charmander", "Grass", 1.1, 1.1, false)
+
+        val pokemonList = listOf(pikachu, bulbasaur, charmander)
+        val data = MutableLiveData(pokemonList)
+
+        whenever(favoritesViewModel.getFavorites()).thenReturn(data)
```
Comment (filbabic) [inline Homework/Week9/app/src/test/java/com/android/pokedex/viewmodel/FavoritesViewModelTest.kt:53]: And as before, great use of `whenever` from Mockito! :]

File: Homework/Week9/app/src/test/java/com/android/pokedex/viewmodel/HomeViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        val pokemonList = listOf(pikachu, bulbasaur, charmander)
+        val data = MutableLiveData(pokemonList)
+
+        whenever(homeViewModel.getAllPokemon()).thenReturn(data)
+        homeViewModel.getAllPokemon().observeForever(observer)
+
+        verify(repository).getAllPokemon()
+        verify(observer).onChanged(pokemonList)
+        assertEquals(data.value, homeViewModel.getAllPokemon().value)
+        verify(repository, times(2)).getAllPokemon()
```
Comment (filbabic) [inline Homework/Week9/app/src/test/java/com/android/pokedex/viewmodel/HomeViewModelTest.kt:65]: Nicely done here!

I think you could've just extracted the value from the first call of `homeViewModel.getAllPokemon()` and then you wouldn't have two interactions here!

File: Homework/Week9/app/src/test/java/com/android/pokedex/viewmodel/LoginViewModelTest.kt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Test
+    fun checkAuthentication() {
+        loginViewModel.checkAuthentication("","")
+        loginViewModel.isAuthenticated.observeForever(observer)
+        verify(observer).onChanged(false)
+
+        loginViewModel.checkAuthentication("validName","validPass")
+        verify(observer).onChanged(true)
```
Comment (filbabic) [inline Homework/Week9/app/src/test/java/com/android/pokedex/viewmodel/LoginViewModelTest.kt:43]: Awesome!

**Comment by abunur** [comment]
Hi Ben, I would also delete the example Unit Test and other sample tests in the application. Those are place holders to help you get started, but it's nice to clean up and get rid of them now that you don't need them.

**Pull request merged**